src/api/c++/z3++.h

```diff
@@ -425,8 +474,17 @@ namespace z3 {
         Z3_symbol m_sym;
     public:
         symbol(context & c, Z3_symbol s):object(c), m_sym(s) {}
-        symbol(symbol const & s):object(s), m_sym(s.m_sym) {}
-        symbol & operator=(symbol const & s) { m_ctx = s.m_ctx; m_sym = s.m_sym; return *this; }
+        symbol(symbol const & s) = default;
+        symbol(symbol && s) noexcept : object{std::forward<object>(s)}, m_sym{std::move(s.m_sym)} {
```

Comment:
useless changes

Reply:
Adding noexcept changes the function signature / where it can be use safely, and change behavior as they affect whether an exception will be propogated, or the program terminated; it is also an optimization.

The `object{std::forward<object>(s)}` change was just, in my opinion, good practice as they allow easier maintainability in case the superclass changes at all; but they are functionally identical.

Reply:
more is not always more.

Reply:
let me expand: I've never seen anyone writing code like this nor I'm aware of any coding standard that suggests this.

Reply:
`noexcept` is recommended by isocpp for move semantics, as it allows moves in noexcept contexts. If you mean `std::move` though I can change it. I typically use it for forwarding arguments of different types, since that is the difference between it and move.

Reply:
Actually, I think the rule of zero applies here as well, so I will update this.